### PR TITLE
New splash screen and about dialog

### LINKED ISSUE #3055: FontForge shows very old splash screen
The latest version available in the PPA for Ubuntu trusty looks like this:

screenshot from 2017-04-15 21-26-44

Would look so much less 90s if it was using this look:

fontforge

### PR DESCRIPTION
**Pull request by ctrlcctrlv**
This closes #3055.

This commit does the following:

# Console mode changes

The first line when you run FontForge, which was:

    Copyright (c) 2000-2018 by George Williams. See AUTHORS for Contributors.

Is now:

    Copyright (c) 2000-2019. See AUTHORS for Contributors.

This will no longer need to be updated randomly whenever we remember to do it. I baked the current year in at compile time by editing `configure.ac` and adding a new variable for it. Why did I remove George's name? He is the main contributor but hasn't been around for seven years. It's time for him to become just another contributor in AUTHORS.

# Splash screen changes

I only made the changes for which there was consensus in #3055. I:

* Changed the image to a newer one which I (previously @probonopd, but I changed it in  made.
* Made the Git hash use a monospace font so it's small enough to fit in one line.
* Changed references which made it seem like this is George's software alone. I love the story of this project, you all know that, I love FontForge history, I run pfaedit.org, I've been trying to get @davelab6 to talk with me on video to flesh out more of the history for my website, so I was very respectful of George while editing the text, making very conservative edits to the history in the about dialog.

### Previously, though cancelled due to consensus

# How I changed it

In case this needs to be done again, I used `gdb` to change the image. First, I loaded a BMP image in `fontforge_main` (startui.c) and set a breakpoint right after loading it. I just made a `GImage` containing the output of `GImageReadBmp("/tmp/ff.bmp")` called `splashimage`.

The following `gdb` settings make things a lot easier:

* set pagination off
* set max-value-size unlimited
* set print elements 0
* set print repeats 0

Then I issued command:
```C
p/x *((*splashimage).u.image.data)@568500
```
You have to either catch the output by telling `gdb` it needs to log...or you can just [set your console to unlimited scrollback and copy it](https://twitter.com/HW_BEAT_THAT/status/1145609148307951616).

I got the number 568500 by multiplying two numbers together found in the struct:

```gdb
(gdb) p (*(*splashimage).u.image)
$46 = {image_type = it_true, delay = 0, width = 379, height = 375, bytes_per_line = 1516, 
  data = 0x15554f5b6010 "\336\236O", clut = 0x0, trans = 4294967295}
```

568500 = 1516 &times; 379

I then pasted all the values into `splashimage.c` as needed. Done and dusted.

### CODE REVIEW

**Comment by probonopd** [comment]
Jay! Thank you so much. Hopefully this can get merged soon.

**Comment by probonopd** [comment]
I wonder why the application needs the image in such a strange format. Why can't it just display a normal png?

**Comment by frank-trampe** [comment]
Good work sorting out the encoding.

I don't really like using a partial crop of the front page of the website like this. It looks sort of awkward. We could scale it down to fit, perhaps.

It also occurred to me, though, that we could just update the old splash screen. I removed the old text, cleaned some edges, redrew the path, and superimposed the new logo and logotype (in bold Merriweather since it's small). What do you think of this?

Logo_Update_Study_20190701_1_1

[Logo_Update_Study_20190701_1_1_xcf.zip](https://github.com/fontforge/fontforge/files/3347100/Logo_Update_Study_20190701_1_1_xcf.zip)

**Comment by frank-trampe** [comment]
Here's a concept for using the website artwork.

Website_Adaptation_Study_20190701_1_1

[Website_Adaptation_Study_20190701_1_1_xcf.zip](https://github.com/fontforge/fontforge/files/3347169/Website_Adaptation_Study_20190701_1_1_xcf.zip)

**Comment by frank-trampe** [comment]
Reposting the concept for the legacy-based design at reduced scale to make comparison easier.

Logo_Update_Study_20190701_1_1

**Comment by probonopd** [comment]
I just plain dislike the [Moiré pattern](https://en.wikipedia.org/wiki/Moir%C3%A9_pattern) that happens when you scan a printed page - looks unprofessional to me.

**Comment by ctrlcctrlv** [comment]
I personally don't care what image is used and am happy to replace the image with whatever there is consensus for. I mostly did this to remove the ©2000-2012 from the splash image. 

In regards to the image's format and the esoteric (in 2019) way it needs to be replaced, that's just down to FontForge's advanced age. Lots of old software works in a similar way. For example Linux distributions still support the X Bitmap format, which is another C code based image format. Having the graphics as part of the binary made the most sense, and including them was rarely done at compile time.

**Comment by ctrlcctrlv** [comment]
(The format also isn't THAT strange, either every four bytes is one pixel, or if the clut is used every byte is one pixel referring to an index in the clut.)

**Comment by probonopd** [comment]
Everything but png and svg is "strange" at this point ;-)

**Comment by frank-trampe** [comment]
@probonopd, I think that most of the visual degradation is from the original halftone in the scanned page. There are also the additional problems of JPEG artifacts and the image originating at a lower resolution than used. But I don't think that that is necessarily bad. The narrative of the image is transitioning from a world of fuzzy analogness to crisp digitalness.

**Comment by ctrlcctrlv** [comment]
I actually have a few fonts I made in FontForge from scans of books/illuminated manuscripts so I can't help but feel some bias for the old image, but will refrain from voting

**Comment by ctrlcctrlv** [comment]
@frank-trampe Let me know what you want to do here. I can push a commit with the image in https://github.com/fontforge/fontforge/pull/3754#issuecomment-507407292 if you'll merge that.

**Comment by frank-trampe** [comment]
@ctrlcctrlv, do you happen to know whether the original image from George Sr.'s book?

**Comment by frank-trampe** [comment]
And another inquiry. The struct that you posted above shows the dimensions as 379 by 375, but the image that you posted is twice that size with features that are clearly one pixel in size (such that it doesn't seem to have been scaled up). Any idea how that happened?

**Comment by frank-trampe** [comment]
I think that most of us would prefer to preserve the old artwork. Let's go in that direction. I have made a few adjustments, although I'm not sure how to deal with the scale issue.

Logo_Update_Study_20190701_1_2

**Comment by frank-trampe** [comment]
And here's the xcf for future adjustment.
[Logo_Update_Study_20190701_1_2_xcf.zip](https://github.com/fontforge/fontforge/files/3348215/Logo_Update_Study_20190701_1_2_xcf.zip)

File: fontforgeexe/startui.c [hunk 1]
```diff
@@ -614,7 +624,11 @@ static int splash_e_h(GWindow gw, GEvent *event) {
 	GDrawSetFont(gw,splash_font);
 	y = splashimage.u.image->height + as + fh/2;
 	for ( i=1; i<linecnt; ++i ) {
-	    if ( is>=lines[i-1]+1 && is<lines[i] ) {
+        if (i == 11) {
```
Comment (skef) [inline fontforgeexe/startui.c]: Can this 11 be expressed more symbolically? Not in the sense of a `#define` for its own sake, but could there be a define (or whatever) for this expressed in more abstract terms? How do we know it's 11 and can that knowledge be expressed in other terms?

Comment (ctrlcctrlv) [reply fontforgeexe/startui.c]: Well the way lines are added is a bit of a mess. Anyone who wants to dramatically change the text is going to have to deal with this. I can put a `#define` and a comment about how to compute it if you think it would help though

**Comment by skef** [comment]
The splash image apparently had special typographical significance for GW given that the website variant of it has the alt tag "Gaudiamus ligature!". I assume that's why he chose it, but I'm not sure it's necessary to preserve that property. If it isn't, another option would be to grab a high res scan of some other illuminated manuscript and fiddle with it similarly. (I agree that the posterization is distracting and less "professional looking"

File: fontforgeexe/startui.c [hunk 2]
```diff
@@ -229,7 +229,7 @@ static void SplashLayout() {
     extern const char *source_modtime_str;
     extern const char *source_version_str;
 
-    uc_strcpy(msg, "When my father finished his book on Renaissance printing (The Craft of Printing and the Publication of Shakespeare's Works) he told me that I would have to write the chapter on computer typography. This is my attempt to do so.");
+    u_strcpy(msg, utf82u_copy("When George Williams’ father finished his book on Renaissance printing (The Craft of Printing and the Publication of Shakespeare's Works) he told him that he would have to write the chapter on computer typography. PfaEdit, later renamed FontForge, was his attempt to do so."));
```
[the remark this answers is not in the document]
Comment (frank-trampe) [reply fontforgeexe/startui.c]: I have not read the book. Is PfaEdit actually part of it in some way? [This page](https://georgeruns.wordpress.com/about/) suggests not.

File: fontforgeexe/startui.c (hunk 1 above)
Comment (skef) [reply fontforgeexe/startui.c]: Even just a comment explaining where it comes from and under what circumstances it would need to change would be sufficient, I think. I'm old fashioned in the way that worries about specific numbers appearing in if guards without some tie to what motivates them.

File: fontforgeexe/startui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (skef) [reply fontforgeexe/startui.c]: I've always read these two sentences as an awkwardly phrased joke. The father asked for a chapter, the result was a program. In my rephrasing I tried to preserve the joke in a less awkward form.

[the remark this answers is not in the document]
Comment (frank-trampe) [reply fontforgeexe/startui.c]: Hmm. I never really read it that way, but the decades between the publication of the book and the completion of FontForge would seem to affirm a less than serious intent as you suggest. With the rewrite, I think that it's also awkward to refer to one George Walton Williams as "George Williams' father". Let me figure out what their respective numbers are. I think that our George is George V and his father George IV, but I'm not sure. Something like this?

"As he drew closer to completing his book on Renaissance printing (...), George Williams IV asked his son, George Williams V, to write a chapter on computer typography. The program PfaEdit, now called FontForge, was his response."

**Comment by ctrlcctrlv** [comment]
@skef OK, here you go.

FontForgeLogo

This is from a font I'm working on. Source: https://twitter.com/HW_BEAT_THAT/status/1100021303996600320

I used FontForge to make the raster Q. I'd be honored to have my work on the splash screen, so I can't help but throw my hat in the ring.

What do you think @frank-trampe , @probonopd ?

**Comment by ctrlcctrlv** [comment]
By the way, just a comment, if we do end up using George's still, his tracing of that `n` is horrible&mdash;it looks like it was done by Potrace, so we might want to fix that...if we don't just use my Q. (Please?)

**Comment by frank-trampe** [comment]
I like yours. There's something poetic about retracing George's steps like that but with better splines.

As for George's tracing of the n, it is almost certainly from potrace since there are clearly more points than he indicated with the red dots.

I'll try to stir up some participation here tomorrow so that we can nail something down this week.

File: fontforgeexe/startui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (ctrlcctrlv) [reply fontforgeexe/startui.c]: I've also read it as a joke. 

* George's father knew nothing about computer typography, and only focused on 17th century bookmaking (which is in itself interesting, don't get me wrong).

* George's father knew computer typography was up-and-coming, and knew his son was good with computers and presumably also interested in typography, so told him he'd have to write the chapter on computer typography.

* George responded by writing PfaEdit in the 90's, which he renamed to FontForge in 2000.

[the remark this answers is not in the document]
Comment (ctrlcctrlv) [reply fontforgeexe/startui.c]: I've never read it as George's father telling George he needed to literally write a chapter in the book&mdash;FontForge is _symbolically_ and _sentimentally_ an extension of his father's work, but it's not _literally_ a chapter in _The Craft of Printing and the Publication of Shakespeare's Works_.

**Comment by FloraCanou** [comment]
> FFLogo
> 
> This is from a font I'm working on. Source: https://twitter.com/HW_BEAT_THAT/status/1100021303996600320

This is definitely favorable. But in the pic your _i_ is different from the printed material's. Maybe consider changing it to match the printed one? (I mean this pic only, nothing to do with the font.) Another suggestion is lowering the saturation of the printed material, since it's clearly oversaturated with something like color stripes now.

**Comment by ctrlcctrlv** [comment]
@FloraCanou Actually it is, the book is handwritten, not every `i` is the same. I do&mdash;however&mdash;see why people might think that, and if you still want me to change it despite this evidence, I'll see about doing so.



As far as the image goes, here's an original. If you can fix it up better than me, by all means have at it! :smile:



This image is cropped, but the original was losslessly extracted by evince from the PDF mentioned in my tweet.

**Comment by ctrlcctrlv** [comment]
(actually, on further inspection, I think I see what you mean. You want me to fix the top part of the `i`, right?, to make it slanted instead of straight)

**Comment by FloraCanou** [comment]
Yes I mean the top of it. Also it's a bit thicker and higher than the _s_(?). 
Now I see part of the colored stripes is from the back of the page. And the other part may have been generated in the processing. I can't download images from Twitter so...

**Comment by ctrlcctrlv** [comment]
> I can't download images from Twitter 

The [one I just uploaded to Github](https://user-images.githubusercontent.com/838783/60515165-a03d0500-9d0d-11e9-98d8-c2d559ced368.png) should have everything you need if you want to try your own processing.

**Comment by frank-trampe** [comment]
I don't think that there's anything wrong with artificially saturating an image if there are no skintones in it, and the image is often more visually engaging for it.

**Comment by skef** [comment]
@ctrlcctrlv There is room for tinkering and improvement (some along lines already pointed out), but I want to say that that your first mockup was already better than I was hoping for when I made my suggestion. Great source material and great modelling. 

If the whole deal could be a bit wider without screwing everything else up, I think the "point" would be a bit stronger if the image was cropped after the second "i" and the first "i" was also made an outline. That gives a bit more of the "during" and after" to balance with the "before".

**Comment by frank-trampe** [comment]
If we're going to rework the canvas, I would also suggest rotating to maximize the orthagonality of the lines.

File: fontforgeexe/startui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (skef) [reply fontforgeexe/startui.c]: If you soften my "requested" to "suggested" (making whatever other appropriate changes) it preserves the basic joke structure but implies the Wilson's were self-aware about it.

**Comment by probonopd** [comment]
It may be a matter of personal preference, but I find the graphics from the website much more adequate to a modern, professional font editing tool. I would go with the graphics from the website.

File: fontforgeexe/startui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (frank-trampe) [reply fontforgeexe/startui.c]: I have confirmed the numbers as [IV](https://twitter.com/shakespearectr/status/393430903705595904) and [V](https://georgeruns.wordpress.com/page/37/?archives-list=1) and have incorporated @skef's suggested change in wording.

"As he drew closer to completing his book on Renaissance printing (...), George Williams IV suggested that his son, George Williams V, write a chapter on computer typography. The program PfaEdit, now called FontForge, was his response."

**Comment by skef** [comment]
The sort of person who finds comfort and security in the graphical style of the website is not going to find that comfort and security in FontForge itself, as it presently stands. Best to let the quirkiness of the project show through from the start (if you ask me).

**Comment by frank-trampe** [comment]
I agree with @skef on that. Our aspirations for a Python-driven FontForge with a new widget kit that drove the design of the present website have not been realized, and the need to support so many legacy workflows and platforms probably mean that that is unlikely to change. FontForge is stable and fast, but it is distinctly not modern.

File: fontforgeexe/startui.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (skef) [reply fontforgeexe/startui.c]: Seems good to me.

**Comment by probonopd** [comment]
Isn't using modern graphics a first step towards making the whole UX more modern?

**Comment by frank-trampe** [comment]
In some cases, yes. But, in this case, the graphic in question is one that shows an interface that differs from the one present in FontForge and includes a feature (editing glyphs in context) that FontForge lacks. It's not absurd, but it is a little awkward. Also, since a substantial modification of the text is off of the table right now, there is an upside to having a graphic that complements the narrative in the text.

**Comment by TotoCoelo** [comment]
Here is my attempt at creating the image for the about dialog. It still looks a bit sketchy but I am trying to accomplish the following.

- the image is wider than the previous example
- the image has been rotated so that the lines are horizontal
- 3rd line is in the process of being digitized, for the time being I copied @ctrlcctrlv glyphs, but this can definately be improved
- below the image the _story_ of fontforge can be told

I hope this might give some ideas.

about_02a

**Comment by frank-trampe** [comment]
@TotoCoelo, I still think that @ctrlcctrlv's work is a good starting point. Regarding your proposal...
* Making the image wider would require making the window wider, which is likely to open a small can of additional worms.
* The outlines really need to line up to the underlying manuscript.
* I like preserving the narrative of the original image, which has manuscript fading to drawn shapes with control points, and then giving way to digital print.
* The extra saturation in @ctrlcctrlv's draft also makes the image a lot more engaging in my opinion.

**Comment by probonopd** [comment]
Really, I don't want to discourage anyone, but the graphics on the website look so much more... professional. Who made them? Could we get that person to design the About box?

**Comment by frank-trampe** [comment]
@davelab6, do you know who designed the current branding and visuals?

@probonopd, we can revisit the possibility of something even better in the future, but the immediate objective is to solve the problem of the outdated branding and copyright information in a way that represents a step forward. The options at hand allow us to do that, and I would like to see this completed soon so that we can move on to other work.

**Comment by frank-trampe** [comment]
Having a stable project mandates defaulting to the least-change approach whenever there is significant cost/benefit uncertainty or disagreement. Given that a lot of us wish to preserve the illuminated manuscript narrative, @ctrlcctrlv's concept is the clear way forward for this effort. Let's make specific suggestions to polish it so that we can get it merged.

I am compiling the suggestions that seem fitting to me and adding one.
* Rotate to make the lines as horizontal as possible.
* Include the second "i" in the image window and make the first one an outline.
* Use the pointy-top "i" to match the row above.
* Use a wider radius on the circular gradient.

**Comment by ctrlcctrlv** [comment]
Thanks for the support Frank. I'll fix those things and have a new version for you within the next twelve hours.

**Comment by skef** [comment]
> The splash image apparently had special typographical significance for GW given that the website variant of it has the alt tag "Gaudiamus ligature!".

Ah, this is a pun on "gaudeamus igitur" -- "so let us rejoice" (apparently).

**Comment by ctrlcctrlv** [comment]
All right, I'm working on this now, making a final render for us. Here's my new, rotated, sharper, cleaned up scan which will be the base:

[FF2.zip](https://github.com/fontforge/fontforge/files/3354380/FF2.zip)

2019-07-03-175115_3840x2160_scrot


I'm so honored to have been chosen to do this!

By the way @probonopd, even though you &ldquo;don't want to discourage anyone,&rdquo; your comments are a great discouragement to me. Instead of trying to help improve the new splash screen demo, you just want to tear it down and replace it with something &laquo;modern&raquo;. 

I've decided to change my vote based on your comments&hellip;**I categorically reject any splash screen image that doesn't keep the tradition of illuminated manuscripts and showing the work of FontForge users.**

So I reject my own commit, https://github.com/fontforge/fontforge/pull/3754/commits/6cd87882b13302ef948e4c8ca0b52c5a8e52906e. How about them apples? :wink:

**Comment by frank-trampe** [comment]
@ctrlcctrlv, I'm excited to see it!

Although I would have chosen different words, I really don't think that @probonopd intended to be combative or personally critical. (@probonopd, correct me if I'm wrong.)

**Comment by ctrlcctrlv** [comment]
FINAL

@frank-trampe Right on cue! Here you go, it's done. I'm converting it into splashimage.c format right now.

**Comment by ctrlcctrlv** [comment]
By the way&mdash;these letters&mdash;were made entirely in FontForge using the (buggy, although I love it) &laquo;Expand Stroke&raquo; tool, and then I made some corrections just using the regular drawing tools&hellip;so this is, how do you say&hellip;&laquo;authentic&raquo;!

**Comment by frank-trampe** [comment]
It's true to the original narrative, it's true to FontForge's character, and it looks good too. In my opinion, at least.

**Comment by ctrlcctrlv** [comment]
I fixed the rotation, the glyph widths, the size of the circular layer mask, and the pointiness of letter `i`.

The font I'm working on based on this book is full of discretionary ligatures and alternates! :-)

**Comment by frank-trampe** [comment]
I figured there might be an alternate "i" in there somewhere.

**Comment by ctrlcctrlv** [comment]
This is what it looks like as of https://github.com/fontforge/fontforge/pull/3754/commits/65810a91456072823491fc56903a277b3e2f5846.

In my opinion, it's ready to be merged.

**Comment by frank-trampe** [comment]
@ctrlcctrlv, it looks like you took out the code to allow a start-up splash?

**Comment by ctrlcctrlv** [comment]
@frank-trampe No I didn't. I only removed some code which caused that weird window resizing effect George added.

**Comment by ctrlcctrlv** [comment]
Splash still appears on start up if settings so dictate. I just took out code that no longer makes sense&hellip;and perhaps never did, that resizing effect was always IMO ugly

**Comment by frank-trampe** [comment]
Ah, okay. This all looks fine to me. @skef?

**Comment by skef** [comment]
I built and tested this on my machine partly to check the most recent changes. @ctrlcctrlv:

1. You mention a "weird window resizing effect" but I think the intended effect of that code was to close the splashscreen automatically after a period of time if it hasn't already been clicked on. That has (generally? always?) worked on my machine before and doesn't work now and is presumably not something we wouldn't remove without discussion. 

2. I don't know if this is due to some bit of configuration fluff hanging around on my machine (although I don't see anything relevant in `prefs`), but in my tests the window comes up sized to the image only, so that you can't see the text below without carefully resizing. You run Arch too so our systems are similar so it's interesting the windows are coming up differently. I assume we want the text visible by default.

**Comment by probonopd** [comment]
At this point I have to wonder whether you would be willing to consider the part of the website graphic that is not showing any UX elements as the graphic for the splash screen:



At least not this ancient book stuff.

**Comment by frank-trampe** [comment]
@probonopd, there is clearly not a consensus for abandoning the current theme at this time. It is unfortunate that there is not a solution here that pleases everybody, but that's a natural consequence of having such a large community. For what it's worth, the splash screen does not show by default on most platforms, so this is rarely the first foot forward for FontForge anyway.

**Comment by khaledhosny** [comment]
> As for George's tracing of the n, it is almost certainly from potrace since there are clearly more points than he indicated with the red dots.

It is just how FontForge's aliased rendering looked like before the introduction of antialiased rendering with Cairo.

**Comment by ctrlcctrlv** [comment]
@skef There are two things to consider here, closely related but not the same, the splash screen and the about screen. The splash screen has never shown the about text and has only shown the image. If it used to disappear... I'm not sure... I don't think so, but let me compare carefully both versions again, from what I saw I only removed the code that resized the window slightly twice to give the appearance of some kind of animation, which doesn't work with the new image as the new image has no copyright year and has the name of the program in a different spot.

**Comment by skef** [comment]
OK, I didn't realize those were separate things. I've checked both and verified the sizing. The initial splash screen not auto-closing is still an issue. 

WRT the current image: Would it be a lot of work to see an option with the logo and "FontForge" closer to centered in the white area? Or alternatively, enlarged to fill more of it? It seems visually pushed to right in this wider format in a way it didn't in your first version.

**Comment by ctrlcctrlv** [comment]
@skef It looks weird to me centered&hellip;it looks better to me right-aligned.

Centered:
FINAL_c

**Comment by frank-trampe** [comment]
@ctrlcctrlv, I think that @skef meant horizontally centered relative to the white circle.

**Comment by ctrlcctrlv** [comment]
Something more like this?

FINAL_c

**Comment by skef** [comment]
Yes, that's what I was thinking of. And seeing it I would vote for that one out of the three.

**Comment by frank-trampe** [comment]
Too far left. (I feel like we're hanging a painting.)

I think that a more important consideration is that the perceptual spacing between the full logo and the bottom and right edges of the image ought to be even.

**Comment by ctrlcctrlv** [comment]
OK, great. By the way, @skef, I did confirm that the image did disappear before and no longer does. Don't know how I missed that. So I'll push &hellip;

&bullet; The image in https://github.com/fontforge/fontforge/pull/3754#issuecomment-508312056
&bullet; Add back `et_timer`, but only to close the splash window, not to do that hacky resize "animation" that no longer makes sense.

**Comment by ctrlcctrlv** [comment]
If that's too far left&hellip;

How about this?

FINAL_c

**Comment by ctrlcctrlv** [comment]
(By the way&mdash;I can also just upload the XCF if that'll help!)

**Comment by frank-trampe** [comment]
I feel ridiculous saying this, but I think that the optimum is half of the way from the most recent one to the one that @skef likes. And now I'm going to stop talking about the spacing, because otherwise I will definitely be ridiculous.

**Comment by skef** [comment]
@ctrlcctrlv sounds good. 

I prefer both of the two most recent versions to the original. This latest one still fixes the "pressed" appearance that was bothering me.

**Comment by ctrlcctrlv** [comment]
@frank-trampe Half-way: 
FINAL_c

I'll use this one :-)

**Comment by frank-trampe** [comment]
Yes!

**Comment by skef** [comment]
Yes that's quite nice.

**Comment by ctrlcctrlv** [comment]
All done. Sorry for the delay, the way I'm overwriting `splashimage.c` isn't exactly fool (or mistake) proof. :-)

File: fontforgeexe/startui.c [hunk 3]
```diff
@@ -304,17 +320,6 @@ void DelayEvent(void (*func)(void *), void *data) {
     GDrawRequestTimer(splashw,100,0,info);
 }
 
-static void DoDelayedEvents(GEvent *event) {
```
Comment (skef) [inline fontforgeexe/startui.c]: `DoDelayedEvents()` is the match for `DelayEvent()`, in the sense that you call the latter to initiate some event at a later time and the former does the initiation. These changes don't remove `DelayEvent()` and it's called in four other files. Therefore with these changes some events that were supposed to happen won't. I've confirmed one of those: At the present branch state if you open a Metrics Window and then choose "Element -> Font Info ..." the window doesn't open.

Probably easiest to just put this function and the else clause back.

File: fontforgeexe/startui.c [hunk 4]
```diff
@@ -391,7 +396,8 @@ static void start_splash_screen(void){
     GDrawSync(NULL);
     GDrawProcessPendingEvents(NULL);
     GDrawProcessPendingEvents(NULL);
-    splasht = GDrawRequestTimer(splashw,1000,1000,NULL);
+
+    splasht = GDrawRequestTimer(splashw,7000,1000,NULL);
```
Comment (skef) [inline fontforgeexe/startui.c]: Not sure why the change to 7000 but it doesn't seem to cause any problems.

File: fontforgeexe/startui.c [hunk 5]
```diff
@@ -632,22 +645,14 @@ static int splash_e_h(GWindow gw, GEvent *event) {
...
-		GDrawResize(gw,splashimage.u.image->width,splashimage.u.image->height-30);
-	    else if ( splash_cnt==2 )
-		GDrawResize(gw,splashimage.u.image->width,splashimage.u.image->height);
-	    else if ( splash_cnt>=7 ) {
-		GGadgetEndPopup();
-		GDrawSetVisible(gw,false);
-		GDrawCancelTimer(splasht);
-		splasht = NULL;
-	    }
-	} else {
```
Comment (skef) [inline fontforgeexe/startui.c]: The aforementioned else clause.

File: fontforgeexe/startui.c (hunk 4 above)
Comment (ctrlcctrlv) [reply fontforgeexe/startui.c]: Because before it was using splash_cnt >=7 to decide when to close the window, as it was doing different things on each timer (it resized on splash_cnt=1, 2 and 3), but now since we only need the timer to fire once I just made it 7000

File: fontforgeexe/startui.c (hunk 5 above)
Comment (ctrlcctrlv) [reply fontforgeexe/startui.c]: Resolved in https://github.com/fontforge/fontforge/pull/3754/commits/caf4608143151aadf8db1e20656a4ac87d8e649a

**Comment by frank-trampe** [comment]
Here we go. The most visible change to FontForge in years!

**Comment by frank-trampe** [comment]
@ctrlcctrlv, it might be worthwhile to post the final layered xcf here so that it is not lost to time.

**Comment by ctrlcctrlv** [comment]
Sure thing, here it is. :smile: 
[FF2_1.zip](https://github.com/fontforge/fontforge/files/3357770/FF2_1.zip)
